libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -268,71 +162,18 @@ const char *AC_AutoTune::type_string() const
         return "Rate D Down";
     case RP_UP:
         return "Rate P Up";
-    case SP_DOWN:
-        return "Angle P Down";
+    case RFF_UP:
+        return "Rate FF Up";
     case SP_UP:
         return "Angle P Up";
-    }
-    return "Bug";
-}
-
-void AC_AutoTune::do_gcs_announcements()
-{
-    const uint32_t now = AP_HAL::millis();
-    if (now - announce_time < AUTOTUNE_ANNOUNCE_INTERVAL_MS) {
-        return;
-    }
-    float tune_rp = 0.0f;
-    float tune_rd = 0.0f;
-    float tune_sp = 0.0f;
-    float tune_accel = 0.0f;
-    char axis_char = '?';
-    switch (axis) {
-    case ROLL:
-        tune_rp = tune_roll_rp;
-        tune_rd = tune_roll_rd;
-        tune_sp = tune_roll_sp;
-        tune_accel = tune_roll_accel;
-        axis_char = 'R';
-        break;
-    case PITCH:
-        tune_rp = tune_pitch_rp;
-        tune_rd = tune_pitch_rd;
-        tune_sp = tune_pitch_sp;
-        tune_accel = tune_pitch_accel;
-        axis_char = 'P';
-        break;
-    case YAW:
-        tune_rp = tune_yaw_rp;
-        tune_rd = tune_yaw_rLPF;
-        tune_sp = tune_yaw_sp;
-        tune_accel = tune_yaw_accel;
-        axis_char = 'Y';
-        break;
-    }
-
-    gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: (%c) %s", axis_char, type_string());
-    send_step_string();
-    if (!is_zero(lean_angle)) {
-        gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: lean=%f target=%f", (double)lean_angle, (double)target_angle);
-    }
-    if (!is_zero(rotation_rate)) {
-        gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: rotation=%f target=%f", (double)(rotation_rate*0.01f), (double)(target_rate*0.01f));
-    }
-    switch (tune_type) {
-    case RD_UP:
-    case RD_DOWN:
-    case RP_UP:
-        gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: p=%f d=%f", (double)tune_rp, (double)tune_rd);
-        break;
     case SP_DOWN:
-    case SP_UP:
-        gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: p=%f accel=%f", (double)tune_sp, (double)tune_accel);
-        break;
+        return "Angle P Down";
+    case MAX_GAINS:
+        return "Find Max Gains";
+    case TUNE_COMPLETE:
+        return "Tune Complete";
     }
-    gcs().send_text(MAV_SEVERITY_INFO, "AutoTune: success %u/%u", counter, AUTOTUNE_SUCCESS_COUNT);
-
-    announce_time = now;
+    return "Bug";
```

Comment:
Not a blocker but I think we should replace "Bug" with "Unknown" and if it should never happen then add a line like this to trigger an internal error.

```
// this should never happen
INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
```

Reply:
Yeah not something I added but  I can make the change.  

Is that all you got ;)

Reply:
@bnsgeyer i'm thinking about giving my IRIS an autotune with this and if it goes OK then I might merge it...

Reply:
Cool. I spoke with Leonard not long ago and he also offered to test it once you were happy with it. Either way works for me.  
what about the parameter issue that you needed to sort out.

Reply:
Incorporated requested change.